### PR TITLE
fix set password & email flow possible to get stuck and onBlur murdering your email

### PR DESCRIPTION
**Pull request by t3chguy**
if you refresh your browser after the SetPassword dialog, whilst in the SetEmail dialog, your password will have been changed but the NagBar will still nag you to change your password and fail every time because the password it remembers is now wrong.
and onBlur of the EditableText murdering the email you have input

### Fixes https://github.com/vector-im/riot-web/issues/6366

### CODE REVIEW

**Comment by dbkr** [comment]
Looks plausible. Assigning @lukebarnard1 to make sure the change to the dispatch is sensible here.

**Comment by t3chguy** [comment]
> GH claims that ChatCreateOrReuseDialog.js is changed in this PR and sprouts a strange error when commenting on that particular part of the diff. Not sure why.


it was changed because that is the dialog that fires the set password&email flow

**Comment by lukebarnard1** [comment]
> make sure the change to the dispatch is sensible here.

Looks better than before, if anything :+1:

**Comment by t3chguy** [comment]
huh I'm not sure why the `ChatCreateOrReuseDialog` changes are in here, I've gotten lost between all the PRs and not sure whether that was intentional, if you feel it is too tangential I can pull it out

**Comment by lukebarnard1** [comment]
>if you feel it is too tangential I can pull it out

Given the changes do not match the commit messages, it'd be best.

**Comment by t3chguy** [comment]
I redid the commits with basically the same content

**Comment by t3chguy** [comment]
(github's diff was a little weird, tried to unstick it)

File: src/components/views/dialogs/ChatCreateOrReuseDialog.js [hunk 1]
```diff
@@ -169,15 +175,15 @@ export default class ChatCreateOrReuseDialog extends React.Component {
                     </p>
                     { profile }
                 </div>
-                <DialogButtons primaryButton={_t('Start Chatting')}
-                    onPrimaryButtonClick={this.props.onNewDMClick} focus="true" />
+                <DialogButtons primaryButton={_t('Start Chatting')} onCancel={this.onCancel}
```
Comment (lukebarnard1) [inline src/components/views/dialogs/ChatCreateOrReuseDialog.js]: This is a functional change, not delint.

**Comment by t3chguy** [comment]
@lukebarnard1 fixed that for you

**Pull request merged**